File: python/Graph.py
class Graph:
    def __init__(self):
        self.nodes = []
        self.adj_list = {}

    def add_node(self, node):
        self.nodes.append(node)
        self.adj_list[node] = []

    def add_edge(self, node1, node2):
        self.adj_list[node1].append(node2)
        self.adj_list[node2].append(node1)


graph = Graph()

graph.add_node('A')
graph.add_node('B')
graph.add_node('C')
graph.add_node('D')
graph.add_node('E')

graph.add_edge('A', 'B')
graph.add_edge('A', 'C')
graph.add_edge('B', 'D')
graph.add_edge('C', 'E')

print(graph.adj_list)

File: javascript/Graph.js
class Graph {
    constructor() {
        this.nodes = []
        this.adjList = {}
    }

    addNode(node) {
        this.nodes.push(node)
        this.adjList[node] = []
    }

    addEdge(node1, node2) {
        this.adjList[node1].push(node2)
        this.adjList[node2].push(node1)
    }
}


const graph = new Graph()

graph.addNode('A')
graph.addNode('B')
graph.addNode('C')
graph.addNode('D')
graph.addNode('E')

graph.addEdge('A', 'B')
graph.addEdge('A', 'C')
graph.addEdge('B', 'D')
graph.addEdge('C', 'E')


console.log(graph.adjList)

